scripts/git_changes_utils_test.py

```diff
@@ -0,0 +1,567 @@
+        ) -> subprocess.Popen[bytes]:  # pylint: disable=unsubscriptable-object
+            if b'remote.origin.url' in cmd_tokens:
+                return process_for_origin_url
+            elif b'remote.upstream.url' in cmd_tokens:
+                return process_for_upstream_url
+            else:
+                return process_for_remote
+        popen_swap = self.swap(subprocess, 'Popen', mock_popen)
+        with popen_swap, self.print_swap:
+            self.assertIsNone(
+                git_changes_utils.get_local_git_repository_remote_name())
+        self.assertTrue(
+            'Warning: Please keep only one remote branch for oppia:develop.\n'
+            'To do that follow these steps:\n'
+            '1. Run the command \'git remote -v\'\n'
+            '2. This command will list the remote references. There will be '
+            'multiple remotes with the main oppia github reopsitory url, but we'
+            ' want to make sure that there is only one main \'upstream\' remote'
+            ' that uses the url https://github.com/oppia/oppia.git. Please use '
+            'the command, \'git remote remove <remote_name>\' on all remotes '
+            'that have the url https://github.com/oppia/oppia.git except for '
+            'the main \'upstream\' remote.\n' in self.print_arr)
+
+    def test_git_diff_name_status_without_error(self) -> None:
+        def mock_start_subprocess_for_result(
+            unused_cmd_tokens: List[str]
+        ) -> Tuple[bytes, None]:
+            return (b'M\tfile1\nA\tfile2', None)
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap:
+            self.assertEqual(
+                git_changes_utils.git_diff_name_status(
+                    'left', 'right', diff_filter='filter'),
+                [
+                    git_changes_utils.FileDiff(status=b'M', name=b'file1'),
+                    git_changes_utils.FileDiff(status=b'A', name=b'file2')
+                ]
+            )
+
+    def test_git_diff_name_status_with_error(self) -> None:
+        def mock_start_subprocess_for_result(
+            unused_cmd_tokens: List[str]
+        ) -> Tuple[str, str]:
+            return ('M\tfile1\nA\tfile2', 'test_oppia_error')
+        subprocess_swap = self.swap(
+            common, 'start_subprocess_for_result',
+            mock_start_subprocess_for_result)
+
+        with subprocess_swap, self.assertRaisesRegex(
+            ValueError, 'test_oppia_error'
+        ):
+            git_changes_utils.git_diff_name_status(
+                'left', 'right', diff_filter='filter')
+
+    def test_git_diff_name_status_with_no_left_and_right(self) -> None:
+        def mock_start_subprocess_for_result(
+            cmd_tokens: List[str]
+        ) -> Tuple[bytes, None]:
+            if cmd_tokens == ['git', 'diff', '--name-status']:
+                return (b'M\tfile1\nA\tfile2', None)
```

Comment:
Using `swap_with_checks` is the more canonical way to assert that this function is called correctly

Reply:
Done.